compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -58,6 +60,18 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
 
     Serializer.serialize(coverage, outputPath, ctx.settings.sourceroot.value)
 
+  private def isClassIncluded(sym: Symbol)(using Context): Boolean =
```

Comment:
I'm wondering about the cost of the pattern matching for each package and classlike. Would it be better to cache the results, for instance for packages (since each package will probably contain many files and declarations)?

At the very least, the `Pattern`s could be created only once, that is, `ctx.settings.coverageExcludeClasslikes.value.map(_.r.pattern)` could be stored and re-used (same thing for coverageExcludeFiles).

Reply:
Ok, added the pre-computing of `Pattern`s.

Not sure if we can add caching in a straightforward way. The easy part of caching is the negative case, which we already short-circuit.